## WHILE LOOPING STATEMWNTS

#### 1. Write a Program to print Series of numbers from 1 to 10.

In [1]:
num=int(input()) #take the input from user as 10 given in the questions
start=1 # initially take the start=1 becuase we need to generate the series from 1 to 10
while start<=num: # in while condition we need to generate the series from 1 to 10 and Check the condition whether number is greater than 0 or not , 
                  # if it is true then it will enter to the whike loop 
    print(start) #print the number every time when the condition true , print the number,where the number is printing on different data 
                 # on each iteration value will be chnange
    start+=1 # we need to increment the start value because to get the number sseries one after and to stop the while loop by increment by 1

 10


1
2
3
4
5
6
7
8
9
10


#### 2. WAP to print series from 5 to 1

In [3]:
start=5 # initially take the start=5 becuase we need to generate the series from 5 to 1 in decending order
while start>0:  # in while condition we need to generate the series from 5 to 1 and Check the condition whether number is greater than 0 or not , 
                # if it is true then it will enter to the whike loop 
    print(start) #print the number every time when the condition true , print the number,where the number is printing on different data 
                 # on each iteration value will be chnange
    start-=1 # we need to increment the start value because to get the number sseries one after and to stop the while loop by decrement by 1

5
4
3
2
1


#### 3. WAP to print series of alphabets from Capital A to E

In [5]:
ch='A' #initially take the character from "A"
while ch<='E': # Check the condition whether character is less than or equal to E or not
    print(ch) # print character if the while condition is true
    ch=chr(ord(ch)+1) # take the ascii value of ch then increment by 1 then we will get next asciee value of character 
                       # that ascii value of character should convert to character and update the ch

A
B
C
D
E


#### 4. WAP to print the numbers which are divisble by 3 from 1 to 7

In [6]:
start=1  #initially take the start value as 1  
while start<=7: # check the condition whether start value is less than or eual to 7 or not, if true enter to while loop
    if start%3==0: # check that start number is divisible by 3 or not, if divisible then it will enter to the if TSB
        print(start) # print if the above condtion true
    start+=1 # increamennt the start value by 1 so in order to stop at certain conditions

3
6


#### 5. WAP to print the numbers which are divisible by both 3 and 5 from 1 to 15

In [10]:
start=1  #initially take the start value as 1  
while start<=15: # check the condition whether start value is less than or eual to 15 or not, if true enter to while loop
    if start%3==0 and start%5==0: # check that start number is divisible by 3 and 5 both or not, if divisible then it will enter to the if TSB
        print(start) # print if the above condtion true
    start+=1 # increamennt the start value by 1 so in order to stop at certain conditions

15


#### 6. WAP to print divisors of 6 from 1 to 5

In [13]:
start=1  #initially take the start value as 1  
num=6  #initially take the number value as 6   
while start<=5: # check the condition whether start value is less than or eual to 5 or not, if true enter to while loop
    if num%start==0: # check that number is divible by start or not, if divisible then it will enter to the if TSB
        print(start) # print if the above condtion true
    start+=1  # increamennt the start value by 1 so in order to stop at certain conditions

1
2
3


#### 7. WAP to print divisor of a given number

In [14]:
start=1  #initially take the start value as 1  
num=int(input())  #take the number from user, for example i will consider num value as 9  
while start<=num: # check the condition whether start value is less than or eual to number or not, if true enter to while loop
    if num%start==0: # check that number is divible by start or not, if divisible then it will enter to the if TSB
        print(start) # print if the above condtion true
    start+=1  # increamennt the start value by 1 so in order to stop at certain conditions

 9


1
3
9


#### 8. WAP to print Odd divisors of a given number

In [19]:
start=1  #initially take the start value as 1  
num=int(input())  #take the number from user, for example i will consider num value as 18 
while start<=num: # check the condition whether start value is less than or eual to number or not, if true enter to while loop
    if start%2!=0: # check that start value is not divisible by 2 or not, if not divisible then it will enter to the if TSB
        if num%start==0: # check that number is divible by start or not, if divisible then it will enter to the if TSB
            print(start) # print if the above condtion true
    start+=1  # increamennt the start value by 1 so in order to stop at certain conditions

 18


1
3
9


#### 9. WAP to count and print the total numbers which are divisible by 4, from 1 to 9

In [20]:
start=1  #initially take the start value as 1  
count=0  #take the count is zero always, becuase counter starts from 0 itself 
while start<=9: # check the condition whether start value is less than or eual to 9 or not, if true enter to while loop
    if start%4==0: # check that start value is divisible by 4 or not, if divisible then it will enter to the if TSB
        count+=1 # update the count from 0 to 1
    start+=1  # increamennt the start value by 1 so in order to stop at certain condition
print(count) #print the total count from 1 to 9

2


#### 10. WAP to check the count of the divisors of a given number is equals to 2 or not. If it is equals then print the prime number, else print not a prime number

In [29]:
start=1  #initially take the start value as 1 
num=int(input()) #take the number from user, for example i will consider num value as 11
count=0 #take the count is zero always, becuase counter starts from 0 itself
while start<=num: # check the condition whether start value is less than or eual to number or not, if true enter to while loop
    if num%start==0: # check that number is divible by start or not, if divisible then it will enter to the if TSB
        count+=1 # update the count from 0 to 1
    start+=1  # increamennt the start value by 1 so in order to stop at certain condition
print('Count :',count) # print the count 
if count==2: # check count is equals to 2 or not, if it equals to 2 then it will enter to the if TSB
    print('Prime Number') # print prime number if condition is true
else: #if count is not equals to 2, if it not equals to 2 then it will enter to the else TSB
    print('Not a Prime Number') # print not prime number if condition is False

 11


Count : 2
Prime Number


#### 11. WAP to check given number is prime or not

In [31]:
start=1  #initially take the start value as 1 
num=int(input()) #take the number from user, for example i will consider num value as 11
count=0 #take the count is zero always, becuase counter starts from 0 itself
while start<=num: # check the condition whether start value is less than or eual to number or not, if true enter to while loop
    if num%start==0: # check that number is divible by start or not, if divisible then it will enter to the if TSB
        count+=1 # update the count from 0 to 1
    start+=1  # increamennt the start value by 1 so in order to stop at certain condition
if count==2: # check count is equals to 2 or not, if it equals to 2 then it will enter to the if TSB
    print('Prime Number') # print prime number if condition is true
else: #if count is not equals to 2, if it not equals to 2 then it will enter to the else TSB
    print('Not a Prime Number') # print not prime number if condition is False

 11


Prime Number


#### 12. WAP to print sum of the numbers from 1 to 5

In [36]:
start=1 #initially take the start value as 1 
Sum=0 #take initially sum=0 always, becuase to do the sum or any number that starts from 0 itself
while start<=5: # check the condition whether start value is less than or eual to 5 or not, if true enter to while loop
    Sum+=start #update the sum value by adding sum+start
    start+=1 # update the start value by 1
print(Sum) # prnt the total sum of the given number

15


#### 13. WAP to print product of the numbers from 1 to n (Factorial of Number Program)

In [37]:
start=1 #initially take the start value as 1 
n=int(input()) # take input from user, for example i will take 5 
prod=1 #take initially prod=1 always, becuase to do the product operations of any number that starts from 1 itself, anything with zero is always 0
while start<=n: # check the condition whether start value is less than or eual to n or not, if true enter to while loop
    prod*=start #update the prod value by multiplying prod*start
    start+=1 # update the start value by 1
print(prod) # prnt the total sum of the given number

 5


120


#### 14. WAP to check sum of the divisors of a given number is equals to same number or not. If it is then print the Perfect Numbers, Else not a Perfect Number. Note : Do not conside the same number as a divisors, else you will not get result

In [44]:
start=1 #initially take the start value as 1 
num=int(input()) # take input from user, for example i will take 6
Sum=0 #take initially sum=0 always, becuase to do the sum or any number that starts from 0 itself
while start<=num-1: # check the condition whether start value is less than or eual to num-1 or not, if true enter to while loop
    if num%start==0: # check that number is divible by start or not, if divisible then it will enter to the if TSB
        Sum+=start #update the sum value by adding sum+start
    start+=1 # update the start value by 1
print('Sum :',Sum) # print the total sum
if num==Sum: # check that num is equals to sum then it will enter to if TSB
    print('Perfect Number') #Print the Perfect Number
else:  # if num is not  equals to sum then it will enter to Else TSB
    print('Not a Perfect Number') #Print the Not a Perfect Number

 6


Sum : 6
Perfect Number


#### 15. WAP to print the common divisors for 8 and 12 from 1 to 9

In [47]:
start=1  #initially take the start value as 1 
divisors=0 #initially take the divisors value as 0
while start<=9: # check the condition whether start value is less than or eual to 9 or not, if true enter to while loop
    if 8%start==0 and 12%start==0: # check that number 8 is divible by start or not and number 12 is divible by start or not, 
                                   # if divisible then it will enter to the if TSB
        divisors=start #store the start value at divisors
        print(divisors) #print the divisors
    start+=1 #update the start value by 1

1
2
4


#### 16. WAP to print the GCD(Greatest Comman Divisor) of the given two numbers. or HCF(Highest Common Factor)

In [49]:
n1=int(input()) #take the input from user , For example i will take as 8
n2=int(input()) #take the input from user , For example i will take as 12
start=1  #initially take the start value as 1 
divisors=0 #initially take the divisors value as 0
while start<=n1 and start<=n2: # check the condition whether start value is less than or eual to n1 and n2 or not, if true enter to while loop
    if n1%start==0 and n2%start==0: # check that number n1 is divible by start or not and number n2 is divible by start or not, 
                                   # if divisible then it will enter to the if TSB
        divisors=start #store the start value at divisors
    start+=1 #update the start value by 1
print(divisors) #print the divisors

 8
 12


4


#### 17. WAP to print the GCD(Greatest Comman Divisor) of the given three numbers. or HCF(Highest Common Factor)

In [51]:
n1=int(input()) #take the input from user , For example i will take as 8
n2=int(input()) #take the input from user , For example i will take as 12
n3=int(input()) #take the input from user , For example i will take as 24
start=1  #initially take the start value as 1 
divisors=0 #initially take the divisors value as 0
while start<=n1 and start<=n2 and start<=n3: # check the condition whether start value is less than or eual to n1 and n2 and n3 or not, 
                                             # if true enter to while loop
    if n1%start==0 and n2%start==0 and n3%start==0: # check that number n1 is divible by start or not and number n2 is divible by start or not,
                                                     # and n3 also , if divisible then it will enter to the if TSB
        divisors=start #store the start value at divisors
    start+=1 #update the start value by 1
print(divisors) #print the divisors

 8
 12
 24


4


#### 18. WAP to print the LCM(Least Common Multiple) of the given two numbers.

In [104]:
n1=int(input()) #take the input from user , For example i will take as 8
n2=int(input()) #take the input from user , For example i will take as 12
start=1  #initially take the start value as 1 
gcd=0 #initially take the divisors value as 0
while start<=n1 and start<=n2: # check the condition whether start value is less than or eual to n1 and n2 or not, if true enter to while loop
    if n1%start==0 and n2%start==0: # check that number n1 is divible by start or not and number n2 is divible by start or not, 
                                   # if divisible then it will enter to the if TSB
        gcd=start #store the start value at divisors
    start+=1 #update the start value by 1
print(n1*n2//gcd) #print the divisors

 8
 12


24


#### 19. WAP to check sum of the numbers from 1 to n is prime number or not.

In [101]:
start=1 #initially take the start value as 1
n=int(input()) #take the input from user , For example i will take as 7
Sum=0 #take initially sum=0 always, becuase to do the sum or any number that starts from 0 itself
count=0 #take the count is zero always, becuase counter starts from 0 itself
while start<=n: # check the condition whether start value is less than or eual to 5 or not, if true enter to while loop
    Sum+=start #update the sum value by adding sum+start
    start+=1 # update the start value by 1
print('Sum',Sum) # prnt the total sum of the given number
start=1 #initially take the start value as 1
while start<=Sum: # check the condition whether start value is less than or eual to number or not, if true enter to while loop
    if Sum%start==0: # check that number is divible by start or not, if divisible then it will enter to the if TSB
        count+=1 # update the count from 0 to 1
    start+=1  # increamennt the start value by 1 so in order to stop at certain condition
if count==2: # check count is equals to 2 or not, if it equals to 2 then it will enter to the if TSB
    print('Prime Number') # print prime number if condition is true
else: #if count is not equals to 2, if it not equals to 2 then it will enter to the else TSB
    print('Not a Prime Number') # print not prime number if condition is False

 2


Sum 3
Prime Number


## WHILE LOOP WITH NUMBERS PROGRAMS :

#### 1. WAP to print all the digits from a given number

In [2]:
num=int(input())
while num!=0:
    ld=num%10
    print(ld)
    num=num//10

 123456


6
5
4
3
2
1


#### 2. WAP to print the digits which are greater than 3 from a given number

In [2]:
num=int(input())
start=1
while num!=0:
    ld=num%10
    if ld>3:
        print(ld)
    num=num//10

 37387391


9
7
8
7


#### 3. WAP to print all odd digits in a given number

In [3]:
num=int(input())
start=1
while num!=0:
    ld=num%10
    if ld%2!=0:
        print(ld)
    num=num//10

 42642990862672


7
9
9


#### 4. WAP to print the sum of the digits in the given numbers

In [5]:
num=int(input())
start=1
sum=0
while num!=0:
    ld=num%10
    sum+=ld
    num=num//10
print(sum)

 31542


15


#### 5. WAP to print the greatest digit from a given number

In [9]:
num=int(input())
gd=num%10
while num!=0:
    ld=num%10
    if ld>gd:
        gd=ld
    num=num//10
print(gd)

 679873


9


#### 6. WAP to print the difference between greatest digit and smallest digit

In [11]:
num=int(input())
gd=num%10
sd=num%10
while num!=0:
    ld=num%10
    if ld>gd:
        gd=ld
    if ld<sd:
        sd=ld
    num=num//10
print(gd-sd)

 34291


8


#### 7. WAP to check whether sum of digits and product of the digits in a given number is equal or not.If it is equals then print **SPY NUMBER** Else Not a SP NUMBER

### **SPY NUMBER PROGRAM**
#### Take any digits of a number, the sum of the digits are equals to the products of digits
#### **sum** of digits == **product** of digits
#### Example : 123
#### 1+2+3==1*2*3
#### 6==6

In [13]:
num=int(input())
sum=0
prod=1
while num!=0:
    ld=num%10
    sum+=ld
    prod*=ld
    num=num//10
if sum==prod:
    print("SPY NUMBER")
else:
    print("NOT A SPY NUMBER")

 123


SPY NUMBER


#### 8. WAP to check sum of the digits in square of the given number is equals to the same number itself for not. If it is true then print **NEON NUMBER**

### **NEON NUMBER PROGRAM**
#### Take any number and that number should be square and for that **Squared Number** take each digit sum of Squared Number, that should equals to the original number
#### **Square** of the **Number** and do the sum of each **Square Numbers of digit** == **Original** Number
#### Example : num=9
#### sqr=9**2=81
#### 81=> 8 + 1 = 9
#### Original Number == Ssum of digit of Square number
#### 9 == 9

In [15]:
num=int(input())
square=num**2
sum=0
while square!=0:
    ld=square%10
    sum+=ld
    square//=10
if sum==num:
    print("NEON Number")
else:
    print("Not A NEON Number")

 9


NEON Number


#### 9. WAP to print the series of the numbers from 1 to each digit in a given number.

In [19]:
num=int(input())
while num!=0:
    ld=num%10
    start=1
    while start<=ld:
        print(f'For digit {ld}, The series starts from 1 to {ld} : {start}')
        start+=1
    num=num//10
    print()

 35


For digit 5, The series starts from 1 to 5 : 1
For digit 5, The series starts from 1 to 5 : 2
For digit 5, The series starts from 1 to 5 : 3
For digit 5, The series starts from 1 to 5 : 4
For digit 5, The series starts from 1 to 5 : 5

For digit 3, The series starts from 1 to 3 : 1
For digit 3, The series starts from 1 to 3 : 2
For digit 3, The series starts from 1 to 3 : 3



#### 10. WAP to print the sum of the series from 1 to each digit from a given number

In [2]:
num=int(input())
while num!=0:
    ld=num%10
    start=1
    sum=0
    while start<=ld:
        sum+=start
        start+=1
    print(f'For digit {ld}, The sum of series starts from 1 to {ld} : {sum}')
    num//=10

 35


For digit 5, The sum of series starts from 1 to 5 : 15
For digit 3, The sum of series starts from 1 to 3 : 6


#### 11. WAP to print the final sum of the factorial of each digit in a given number (Fact - Product of the series from 1 to n)

### **STRONG NUMBER PROGRAM**
#### Take any number of the digits that should be the **sum** of the **factorial** of the number is equals to the original number
#### **Sum** of the **factorial(Product)** of the number == **Original** Number
#### Example : num=145
#### fact of each digit : 5 - 1*2*3*4*5=120, 4 - 1*2*3*4=24, 1 - 1 = 1
#### sum of fact : 120+24+1=145
#### Original Number == **Sum** of the **factorial(Product)** of the number
#### 145 == 145

In [4]:
num=int(input())
num1=num
sum=0
while num!=0:
    ld=num%10
    start=1
    prod=1
    while start<=ld:
        prod*=start
        start+=1
    num//=10
    sum+=prod
    print(f'For digit {ld}, The Product of series starts from 1 to {ld} : {prod}')
print(f'Sum of all Product is {sum}')
if num1==sum:
    print('Strong Number')
else:
    print('Not a Strong Number')

 145


For digit 5, The Product of series starts from 1 to 5 : 120
For digit 4, The Product of series starts from 1 to 4 : 24
For digit 1, The Product of series starts from 1 to 1 : 1
Sum of all Product is 145
Strong Number


#### 12. WAP to print count of the digits in the given number.

In [5]:
num=int(input())
count=0
while num!=0:
    ld=num%10
    count+=1
    num//=10
print(count)

 124


3


#### 13. WAP to count the Odd Digits from given number

In [6]:
num=int(input())
count=0
while num!=0:
    ld=num%10
    if ld%2!=0:
        count+=1
    num//=10
print(count)

 2379365


5


#### 14. WAP to power each digit with the count of the digits in the given number and print the power.

In [8]:
num=int(input())
count=0
num1=num
while num!=0:
    count+=1
    num//=10
print('Count :',count)
while num1!=0:
    ld=num1%10
    print(ld**count)
    num1//=10

 123


Count : 3
27
8
1


#### 15. WAP to check sum of all power of the digitws with its count of the total digits in the given number. **ARMSTRONG Number**

### **ARMSTRONG NUMBER PROGRAM**
#### Take any Number, the length(count) of the number should be powerd by each digit of the number and do the sum for each powered digit, that should eqauls to the original number.
#### Sum of the each Power digit of the given number length == Original Number
#### Example : num = 153
#### len/count = 153 -> 3
#### 1**3 + 5**3 + 3**3 = 1 + 125 + 27 = 153
#### Original Number == Sum of the each Power digit of the given number length
#### 153 == 153

In [2]:
num=int(input())
count=0
num1=num
while num!=0:
    count+=1
    num//=10
print('Count :',count)
Sum=0
while num1!=0:
    ld=num1%10
    num1//=10
    Sum=Sum+(ld**count)
print(Sum)
if num1==Sum:
    print('Armstrong Number')
else:
    print('Not a Armstrong Number')

 153


Count : 3
153
Armstrong Number
